bindings/matrix-sdk-crypto-ffi/src/machine.rs

```diff
@@ -601,7 +599,7 @@ impl OlmMachine {
             content: &'a RawValue,
         }
 
-        let event: OriginalSyncRoomEncryptedEvent = serde_json::from_str(event)?;
+        let event = Raw::from_json(RawValue::from_string(event.to_string())?);
```

Comment:
I wonder why `clippy::str_to_string` didn't trigger here 🤔 
In any case `RawValue::from_string` only makes sense when there is already a `String` you want to transfer ownership of. I think this should just be

```suggestion
        let event = serde_json::from_str(event)?;
```

(potentially with a type annotation so it's clearer what's being deserialized)